app_dart/lib/src/service/branch_service.dart

```diff
@@ -119,15 +119,15 @@ class BranchService {
   /// Latest beta and stable branches are retrieved based on 'beta' and 'stable' tags. Dev branch is retrived
   /// as the latest flutter candidate branch.
   Future<List<Map<String, String>>> getReleaseBranches(
-      {required gh.GitHub github, required gh.RepositorySlug slug}) async {
-    final String betaSha = (await github.repositories.getBranch(slug, 'beta')).commit!.sha!;
-    final String stableSha = (await github.repositories.getBranch(slug, 'stable')).commit!.sha!;
-
-    List<gh.Branch> branches = await github.repositories.listBranches(slug).toList();
-    final String devName = await _getDevBranch(github: github, slug: slug, branches: branches);
-    final String betaName = branches.where(((gh.Branch b) => b.commit!.sha == betaSha)).single.name!;
-    final String stableName = branches.where(((gh.Branch b) => b.commit!.sha == stableSha)).single.name!;
-
+      {required GithubService githubService, required gh.RepositorySlug slug}) async {
+    List<gh.Branch> branches = await githubService.github.repositories.listBranches(slug).toList();
+    final String devName = await _getDevBranch(github: githubService.github, slug: slug, branches: branches);
+
+    final String betaName =
+        (await githubService.getFileContent(slug, 'bin/internal/release-candidate-branch.version', ref: 'beta')).trim();
```

Comment:
I added a trim() after the `replaceAll` and `decode` operations. Based on experiments with fetch file on [stable through github api](https://api.github.com/repos/flutter/flutter/contents/bin/internal/release-candidate-branch.version?ref=stable) and [beta through github api](https://api.github.com/repos/flutter/flutter/contents/bin/internal/release-candidate-branch.version?ref=beta), the decoded release branch version files would for some reasons have trailing '\n' attached to it, and the trailing '\n' could not be removed by `replaceAll`.  I tested this with my personal repo as well, here is the [api response for my personal repo](https://api.github.com/repos/XilaiZhang/miscellaneous-side-project/contents/push_box/README.md). 

So I ended up adding a trim() at the end.

Reply:
All version files (and part of the Flutter style guide, and generally a widely accepted practice) is to have a trailing newline before EOF

Reply:
Yeah the reason i added trim() following a replaceAll('\n') is related to decoding. I explained in more details in https://github.com/flutter/cocoon/pull/2075#discussion_r960976435.